components/partials/common/trade/order-inputs.vue

```diff
@@ -0,0 +1,535 @@
+      type: Boolean,
+      required: true
+    },
+
+    tradingType: {
+      type: String as PropType<TradeExecutionType>,
+      required: true
+    },
+
+    price: {
+      type: String,
+      required: true
+    },
+
+    amount: {
+      type: String,
+      required: true
+    },
+
+    quoteAmount: {
+      type: String,
+      required: true
+    },
+
+    proportionalPercentage: {
+      type: Number,
+      required: true
+    },
+
+    amountStep: {
+      type: String,
+      required: true
+    },
+
+    lastTradedPrice: {
+      type: Object as PropType<BigNumberInBase>,
+      required: true
+    },
+
+    market: {
+      type: Object as PropType<UiSpotMarketWithToken>,
+      required: true
+    },
+
+    baseAvailableBalance: {
+      type: Object as PropType<BigNumberInBase>,
+      required: true
+    },
+
+    quoteAvailableBalance: {
+      type: Object as PropType<BigNumberInBase>,
+      required: true
+    },
+
+    sells: {
+      type: Array as PropType<UiPriceLevel[]>,
+      required: true
+    },
+
+    buys: {
+      type: Array as PropType<UiPriceLevel[]>,
+      required: true
+    },
+
+    takerFeeRate: {
+      type: Object as PropType<BigNumberInBase>,
+      required: true
+    },
+
+    feeRate: {
+      type: Object as PropType<BigNumberInBase>,
+      required: true
+    },
+
+    hasPrice: {
+      type: Boolean,
+      required: true
+    },
+
+    orderTypeBuy: {
+      type: Boolean,
+      required: true
+    },
+
+    executionPrice: {
+      type: Object as PropType<BigNumberInBase>,
+      required: true
+    },
+
+    hasInputErrors: {
+      type: Boolean,
+      required: true
+    },
+
+    totalWithFees: {
+      type: Object as PropType<BigNumberInBase>,
+      required: true
+    },
+
+    hasAmount: {
+      type: Boolean,
+      required: true
+    },
+
+    hasAdvancedSettingsErrors: {
+      type: Boolean,
+      required: true
+    },
+
+    averagePriceOption: {
+      type: Number as PropType<AveragePriceOptions>,
+      required: true
+    }
+  },
+
+  data() {
+    return {
+      inputBaseAmount: '',
+      inputQuoteAmount: '',
+      inputPrice: '',
+      inputPostOnly: false,
+      inputProportionalPercentage: 0,
+      inputSlippageTolerance: '0.5'
+    }
+  },
+
+  computed: {
+    $percentageOptions(): any {
+      return this.$refs.percentageOptions
+    },
+
+    inputQuoteAmountToBigNumber(): BigNumberInBase {
+      const { inputQuoteAmount } = this
+
+      return new BigNumberInBase(inputQuoteAmount)
+    },
+
+    inputAmountToBigNumber(): BigNumberInBase {
+      const { inputBaseAmount } = this
+
+      return new BigNumberInBase(inputBaseAmount)
+    },
+
+    hasQuoteAmount(): boolean {
+      const { inputQuoteAmountToBigNumber, priceStep } = this
+
+      return (
+        !inputQuoteAmountToBigNumber.isNaN() &&
+        inputQuoteAmountToBigNumber.gt(0) &&
+        inputQuoteAmountToBigNumber.gte(priceStep)
+      )
+    },
+
+    priceStep(): string {
```

Comment:
Let's abstract `amountStep` to this component and abstract the way we format the placeholder values from the decimalPlaces into a function on the `formatter.ts` file.